### DISCUSSION TITLE
Build warnings when including Perception project

### DISCUSSION THREAD

**Discussion by PatrickDanino** [pointfreeco/swift-perception#61] [General]
### Description

New build warnings appeared with the latest update.

In `Bindable`, line 63:
```
Main actor-isolated property 'id' cannot be used to satisfy nonisolated protocol requirement
```

In `Environment`, lines 16, 27 and 46:
```
Cannot form key path that captures non-sendable type 'KeyPath<T, T>'
```

### Checklist

- [X] I have determined that this bug is not reproducible using Swift's observation tools. If the bug is reproducible using the `@Observable` macro or another tool from the `Observation` framework, please [file it directly with Apple](https://github.com/apple/swift/issues/new/choose).
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

### Expected behavior

We're hoping the issues above can be resolved unless they are specific to Swift 5.9 and fixed in Swift 5.10.

### Actual behavior

We generally treat build warnings as errors.

### Steps to reproduce

Clean build a new project that includes Perception

### Perception version information

1.1.5

### Destination operating system

iOS 15

### Xcode version information

15.2

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @PatrickDanino, the `id` warning is due to `@ObservedObject` implicitly "upgrading" the enclosing type to `@MainActor`. I believe this behavior is going away in Swift 6, and the key path one is a bug in Swift that will be fixed in Swift 6.

However, how is your project set up that you are seeing warnings for swift-perception? I created a fresh project with full concurrency warnings, added swift-perception, compiling, and I don't see any warnings. I tried in 15.2 and 15.3, and I also turned on "treat warnings as errors". Can you provide a sample project that demonstrates the behavior you are seeing?

**Comment by mbrandonw** [comment]
Hi @PatrickDanino, I still haven't been able to reproduce this, so I am going to convert this to a discussion for now. Please feel free to continue the conversation over there if you uncover something new.

**Reply by mbrandonw** [reply]
Hi @PatrickDanino, I just want to follow up one last time on this because we plan on making this change in a few of our other libraries. Are you definitely able to reproduce the problem of sendable warnings in this library surfacing in your projects that depend on it?

**Reply by PatrickDanino** [reply]
Sorry for the delayed response, I've been very short on time the last couple of weeks while working on a large project.

It appears one of your recent changes fixed the first issue, though I am still seeing the second. That one is likely due to [this KeyPath issue](https://forums.swift.org/t/sendablekeypath/67195/3). While far from ideal, adding `extension KeyPath: @unchecked Sendable { }` suppresses the warning.

**Reply by mbrandonw** [reply]
Thanks for updating us. Yes, the key path one is a Swift bug that will be fixed in Swift 6. I personally would not conform `KeyPath` to sendable like that because you may forget to remove it. It would probably be better to just ignore those warnings for now.

**Reply by PatrickDanino** [reply]
Understood! We track these types of changes separately from our regular code, so there's little risk we'll forget about it, but that may not be the case for everyone.

To answer one of the previously asked questions, we have the default for `Strict Concurrency Checking` set to `Targeted`. That is likely why a project wouldn't show these types of warnings (errors) by default, but we have this set so we don't ignore them given the size and complexity of our codebase.
